### PR TITLE
[ROCm4.5] Allow setting the default FIND_MODE at CMake configuration phase

### PR DESCRIPTION
**Pull request by JehandadKhan**

### CODE REVIEW

File: include/miopen/config.h.in [hunk 1]
```diff
@@ -73,6 +73,7 @@
 //     to BF16 results. This affects the main functionality of the library.
 #cmakedefine01 MIOPEN_USE_RNE_BFLOAT16
 
+#cmakedefine MIOPEN_DEFAULT_FIND_MODE "@MIOPEN_DEFAULT_FIND_MODE@"
```
Comment (atamazov) [inline include/miopen/config.h.in:76]: I that that if you remove double quotes, then you'll be able to use the macro where the value of `FindMode::Values::Default_` is defined. The value of macro would be an identifier (not literal C string). The changes in find_controls.cpp won't be necessary.

**Review by atamazov — COMMENTED** [review]
Let's remove the default find mode from internal API. It may also worth adding a comment instead, that explains how the user's app can get the default find mode at run time. What do you think?

**Comment by atamazov** [comment]
And we can also add this new functionality to develop, I guess.

**Comment by JehandadKhan** [comment]
> Let's remove the default find mode from internal API. It may also worth adding a comment instead, that explains how the user's app can get the default find mode at run time. What do you think?

I like that idea, let me update the PR

**Comment by JehandadKhan** [comment]
@atamazov Your comments have been addressed, please review.

**Review by atamazov — APPROVED** [review]
LGTM

**Comment by atamazov** [comment]
Let me test & merge it

**Comment by atamazov** [comment]
> And we can also add this new functionality to develop, I guess.

And we'll update CI test stages that require Normal Find mode (change from ENV to CMake) to test this feature.

**Pull request merged**